plugins/TrollManagement/class.trollmanagement.plugin.php

```diff
@@ -469,4 +499,179 @@ private function checkTroll(int $userID, array &$args) {
             $args['IsValid'] = false;
         }
     }
+
+    /**
+     * Add a configuration popup to the Troll Management plugin.
+     *
+     * @param Controller $sender
+     */
+    public function settingsController_trollManagement_create($sender) {
+        // Prevent non-admins from accessing this page
+        $sender->permission('Garden.Settings.Manage');
+
+        $sender->title(sprintf(t('%s settings'), t('Troll Management')));
+        $sender->setData('PluginDescription', $this->getPluginKey('Description'));
+
+        $sender->Form = new Gdn_Form();
+        $validation = new Gdn_Validation();
+        $configurationModel = new Gdn_ConfigurationModel($validation);
+
+        $configurationModel->setField([
+            'TrollManagement.PerFingerPrint.Enabled' => c('TrollManagement.PerFingerPrint.Enabled', false),
+            'TrollManagement.PerFingerPrint.MaxUserAccounts' => c('TrollManagement.PerFingerPrint.MaxUserAccounts', 5),
+        ]);
+
+        $sender->Form->setModel($configurationModel);
+
+        // If seeing the form for the first time...
+        if ($sender->Form->authenticatedPostBack() === false) {
+            $sender->Form->setData($configurationModel->Data);
+        } else {
+            $configurationModel->Validation->applyRule('TrollManagement.PerFingerPrint.Enabled', 'Boolean');
+
+            if ($sender->Form->getFormValue('TrollManagement.PerFingerPrint.Enabled')) {
+                $configurationModel->Validation->applyRule('TrollManagement.PerFingerPrint.MaxUserAccounts', 'Required');
+                $configurationModel->Validation->applyRule('TrollManagement.PerFingerPrint.MaxUserAccounts', 'Integer');
+                $configurationModel->Validation->applyRule(
+                    'TrollManagement.PerFingerPrint.MaxUserAccounts',
+                    'validatePositiveNumber',
+                    sprintf(t('%s must be a positive number.'), t("Maximum user's accounts"))
+                );
+            }
+
+            if ($sender->Form->save()) {
+                $sender->StatusMessage = t('Your changes have been saved.');
+            }
+        }
+
+        $sender->render($this->getView('configuration.php'));
+    }
+
+    /**
+     * Add a _probable_ justification as to why a user's is on the applicant's list if there are too many user
+     * accounts using the same fingerprint.
+     *
+     * @param Controller $sender
+     * @param array $args
+     */
+    public function base_applicantInfo_handler($sender, $args) {
+        if (c('TrollManagement.PerFingerPrint.Enabled', false)) {
+            $maxSiblingAccounts = c('TrollManagement.PerFingerPrint.MaxUserAccounts');
+            $userFingerprint = val('Fingerprint', $args['User']);
+            $fingerprintUsages = $this->getSharedFingerprintsUsersCount($userFingerprint);
```

Comment:
What happens if `$userFingerprint` is `null` and you call `getSharedFingerprintsUsersCounts()`?